Subject: pppd: Fix a stack variable overflow in MSCHAP-v2
Origin: upstream, commit:36733a891fb56594fcee580f667b33a64b990981
From: John Crispin <blogic@openwrt.org>

This fixes a bug introduced in 08ef47ca ("pppd: Make MSCHAP-v2 cope
better with packet loss").

Signed-off-by: Paul Mackerras <paulus@samba.org>

diff --git a/pppd/chap_ms.c b/pppd/chap_ms.c
index 016b42e0c10e..c2bd00f9c6f7 100644
--- a/pppd/chap_ms.c
+++ b/pppd/chap_ms.c
@@ -382,7 +382,7 @@ chapms2_make_response(unsigned char *response, int id, char *our_name,
 		      unsigned char *private)
 {
 	const struct chapms2_response_cache_entry *cache_entry;
-	unsigned char auth_response[MS_AUTH_RESPONSE_LENGTH];
+	unsigned char auth_response[MS_AUTH_RESPONSE_LENGTH+1];
 
 	challenge++;	/* skip length, should be 16 */
 	*response++ = MS_CHAP2_RESPONSE_LEN;
-- 
2.1.0

